syntax = "proto2";

package json2pb.test;

message EchoRequest {
    required string text = 1;
}

message ComplexMessage {
    required string _str = 1;
    optional float  _float = 2;
    repeated sint64 _int = 3;
    required bytes  _bin = 4;
    required bool   _bool = 5;
    message SubMessage {
        required string field = 1;
        repeated EchoRequest echo = 2;
        extensions 100 to 199;
    }
    optional SubMessage sub = 10;
    enum SubEnum {
        VALUE1 = 10;
        VALUE2 = 20;
    };
    repeated SubEnum _enum = 11;
    repeated string str_list = 12;
    optional string str_optional = 13;
    repeated string str_list_ill = 14;

    optional string implicit_string_cast_int = 15;
    optional string implicit_string_cast_double = 16;
    optional string implicit_string_cast_boolean = 17;
    optional string implicit_string_cast_array = 18;
    optional string implicit_string_cast_object = 20;
    repeated string implicit_string_cast_repeated_int = 21;

    extensions 100 to 199;
}

extend ComplexMessage {
    optional bool e_bool = 101;
    optional uint32 e_int = 102;
}

enum ValuesEnum
{
	VALUE_1 = 1;
	VALUE_2 = 2;
	VALUE_3 = 3;
	VALUE_4 = 4;
};

message EnumMessage
{
	repeated ValuesEnum enums = 1;
};


// vim: sts=4 sw=4 et
